plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -722,6 +722,11 @@ public function add_image_sizes() {
 		$single            = wc_get_image_size( 'single' );
 		$gallery_thumbnail = wc_get_image_size( 'gallery_thumbnail' );
 
+		// check if string (like `thumbnail`) wasn't passed with `woocommerce_get_image_size_gallery_thumbnail` filter
```

Comment:
I'm not sure why we only do this extra check for `$gallery_thumbnail` (why don't we do the same for `$single`)?

If the problem is that mis-use of the `woocommerce_get_image_size_<SIZE>` hook can lead to `wc_get_image_size()` returning something other than an array, perhaps we should solve the problem from within that function rather than here?